modules/Hoot/ui/modals/importDataset.js

```diff
@@ -499,7 +521,10 @@ export default class ImportDataset {
 
         // overwrite the submit click action with a cancel action
         this.submitButton.on( 'click', () => {
-            Hoot.api.cancelJob(this.jobId);
+            if ( this.jobId ) {
+                Hoot.api.cancelJob(this.jobId);
+            }
+            Hoot.api.cancelUpload( this.source );
```

Comment:
Did we find out if calling this on a completed request throws an error?
If so we could probably put it in an `} else { }` with the `if`

Reply:
this prevents the cancelJob from running without any jobId. I've pushed all the changes you requested. This change will prevent cancelJob from being run if the upload is cancelled prior to the first jobId response. Once that happens, you wont' be able to cancel the upload

Reply:
Right, but what happens if the upload request has completed, there is now a jobId, and the cancel button is clicked.  The hoot cancelJob() will be called but so will cancelUpload() which may throw an error.  Putting cancelUpload in an else here will make it either one or the other.